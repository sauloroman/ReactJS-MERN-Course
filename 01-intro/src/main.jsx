import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import {App} from './App.jsx'

// import './bases/01-const-let.js'
// import './bases/02-template-string.js'
// import './bases/03-objeto-literal.js'
// import './bases/04-arrays.js'
// import './bases/05-funciones.js'
// import './bases/06-deses-obj.js'
// import './bases/07-deses-arr.js'
// import './bases/08-imp-exp.js'
// import './bases/09-promesas.js'
// import './bases/10-fetch.js'
// import './bases/11-async-await.js'
import './bases/12-ternario.js'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <App />
  </StrictMode>,
)
