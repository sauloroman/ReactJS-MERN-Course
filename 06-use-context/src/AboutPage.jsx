import { useContext } from "react"
import { UserContext } from "./context/UserContext"

export const AboutPage = () => {

  const { user } = useContext(UserContext)

  return (
    <div>
      <h1 className="heading__page">About Page</h1>

      <div className="user">
        <p className="user__name">{user.name}</p>
        <p className="user__email">{user.email}</p>
        <p className="user__age">{user.age}</p>
      </div>
    </div>
  )
}
