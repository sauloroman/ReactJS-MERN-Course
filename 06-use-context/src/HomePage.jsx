export const HomePage = () => {
  return (
    <div>
      <h1 className="heading__page">Home Page</h1>
    </div>
  )
}
