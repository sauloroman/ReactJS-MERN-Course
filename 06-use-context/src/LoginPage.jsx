import { useContext } from "react"
import { UserContext } from "./context/UserContext"

export const LoginPage = () => {

  const { setUser } = useContext(UserContext)

  return (
    <div>
      <h1 className="heading__page">Bienvenido a la app</h1>
      <button 
        className="btn"
        onClick={() => setUser({ name: 'Saulo Román', email: 'romansantillan1998@outlook.com', age: 26, })}>Iniciar Sesión</button>
    </div>
  )
}
