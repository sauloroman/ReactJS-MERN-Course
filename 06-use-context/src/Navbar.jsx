import { NavLink } from 'react-router-dom'

export const Navbar = () => {
  return (
    <div className="nav">
      <ul className="nav__list">
        <NavLink to={'/'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>Home</li>
        </NavLink>
        <NavLink to={'/about'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>About</li>
        </NavLink>
        <NavLink to={'/login'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>Login</li>
        </NavLink>
      </ul>
    </div>
  )
}
