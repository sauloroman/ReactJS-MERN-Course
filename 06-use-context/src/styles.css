*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

html {
  font-size: 62.5%;
}

body {
  box-sizing: border-box;
  font-size: 1.6rem;
  font-family: 'Poppins', sans-serif;
  color: #555;
}

.container {
  margin: 0 auto;
  width: 120rem;
  max-width: 95%;
}

.heading {
  font-size: 3.2rem;
  color: #5959ec;
  margin-bottom: 2rem;
}

.heading__page {
  font-size: 2rem;
  color: #000;
}

.nav {
  background-color: #000;
  margin-bottom: 2rem;
}

.nav__list {
  list-style: none;
  display: flex;
  justify-content: space-around;
}

.nav__item {
  width: 100%;
  text-align: center;
  padding: 1rem 0;
  color: #fff;
  text-decoration: none;
}

.btn {
  background-color: #5959ec;
  font-family: inherit;
  color: #fff;
  padding: 1rem 2rem;
  border: none;
  cursor: pointer;
}