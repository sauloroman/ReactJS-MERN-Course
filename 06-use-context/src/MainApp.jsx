import { Navigate, Route, Routes } from "react-router-dom"
import { Navbar } from "./Navbar"
import { HomePage } from "./HomePage"
import { AboutPage } from "./AboutPage"
import { LoginPage } from "./LoginPage"
import { UserProvider } from "./context/UserProvider"

export const MainApp = () => {
  return (
    <UserProvider>
      <div className="container">
        <h1 className="heading">UseContext</h1>
        <Navbar />

        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/about" element={<AboutPage />} />
          <Route path="/login" element={<LoginPage />} />
          <Route path="/*" element={<Navigate to={'/'}/>} />
        </Routes>
      </div>
    </UserProvider>
  )
}
