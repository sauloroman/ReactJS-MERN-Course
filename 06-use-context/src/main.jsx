import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { MainApp } from './MainApp'
import { BrowserRouter } from 'react-router-dom'
import './styles.css'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <BrowserRouter>
      <MainApp />
    </BrowserRouter>
  </StrictMode>,
)
