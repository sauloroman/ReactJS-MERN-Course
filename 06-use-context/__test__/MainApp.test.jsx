import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { MainApp } from '../src/MainApp'

describe('Tests in MainApp Component', () => {

  test('It must match with snapshot', () => {

    const { container } = render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>
    )

    expect( container ).toMatchSnapshot()

  })

  test('It should show Home Page as a initial page', () => {

    render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>
    )

    const titlePage = screen.getByText('Home Page')
    expect( titlePage ).toBeTruthy()

  })

  test('it must show the login page correctly', () => {

    render(
      <MemoryRouter initialEntries={['/login']} >
        <MainApp />
      </MemoryRouter>
    )

    const loginPageTitle = screen.getByText('Bienvenido a la app')
    expect( loginPageTitle ).toBeTruthy()

  })

  test('it must show the about page correctly', () => {

    render(
      <MemoryRouter initialEntries={['/about']}>
        <MainApp />
      </MemoryRouter>
    )

    const aboutPageTitle = screen.getByText('About Page')
    expect( aboutPageTitle ).toBeTruthy()

  })

})