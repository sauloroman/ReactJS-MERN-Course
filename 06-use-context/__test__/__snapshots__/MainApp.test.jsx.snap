// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in MainApp Component It must match with snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="heading"
    >
      UseContext
    </h1>
    <div
      class="nav"
    >
      <ul
        class="nav__list"
      >
        <a
          aria-current="page"
          class="nav__item--active nav__item"
          href="/"
        >
          <li
            class="nav__text"
          >
            Home
          </li>
        </a>
        <a
          class="false nav__item"
          href="/about"
        >
          <li
            class="nav__text"
          >
            About
          </li>
        </a>
        <a
          class="false nav__item"
          href="/login"
        >
          <li
            class="nav__text"
          >
            Login
          </li>
        </a>
      </ul>
    </div>
    <div>
      <h1
        class="heading__page"
      >
        Home Page
      </h1>
    </div>
  </div>
</div>
`;
